core/src/main/java/nl/tudelft/contextproject/core/entities/Colour.java

```diff
@@ -1,21 +1,103 @@
 package nl.tudelft.contextproject.core.entities;
 
 import com.badlogic.gdx.graphics.Color;
-import lombok.Getter;
 
-/**
- * Created by Mitchell on 12-5-2015.
- */
+import java.util.Collection;
+import java.util.Iterator;
+
 public enum Colour {
 
-    RED(255, 0, 0), YELLOW(255, 255, 0), BLUE(0, 0, 255), LIGHTGREEN(0, 255, 0), GREEN(34, 170, 34), GREY(127, 127, 127),
-    DAKBLUE(0, 0, 255), PURPLE(128, 0, 128), PINK(250, 105, 180), BROWN(139, 69, 69), ORANGE(255, 165, 0),
-    SALMON(255, 181, 153), BLACK(0, 0, 0);
+    RED(1, 0, 0, -16776961),
+    YELLOW(1, 1, 0, -65281),
+    BLUE(0, 0, 1, 65535),
+    GREEN(34 / 255f, 170 / 255f, 34 / 255f, 581575423),
+    PURPLE(128 / 255f, 0, 128 / 255f, -2147450625),
+    ORANGE(255 / 255f, 165 / 255f, 0, -5963521),
+    BLACK(0, 0, 0, 255);
+
+    private Color libgdxColor;
+    private int pixelValue;
+
+    /**
+     * Constructor method.
+     * @param r The red value for a colour; lies between 0 and 1.
+     * @param g The green value for a colour; lies between 0 and 1.
+     * @param b The blue value for a colour; lies between 0 and 1.
+     * @param pixelValue The colour value of a pixel from the pixmap.
+     */
+    Colour(float r, float g, float b, int pixelValue) {
+        this.libgdxColor = new Color(r, g, b, 1);
+        this.pixelValue = pixelValue;
+    }
+
+    /**
+     * Method that returns the colour corresponding to the pixelValue.
+     * Checks the Enums as defined at the top of the class.
+     * @param p The pixelValue.
+     * @return The corresponding colour.
+     */
+    public static Colour getColour(int p) {
+        for (Colour c : Colour.values()) {
+            if (p == c.pixelValue) {
+                return c;
+            }
+        }
+        return BLACK;
+    }
+
+    /**
+     * Method that checks if two colours are complementary.
+     * @param colours Collection that contains the two colours that need to be compared.
+     * @return Boolean value.
+     */
+    public static boolean areComplementary(Collection<Colour> colours) {
+        if (colours.size() != 2) {
```

Comment:
These checks (another one in the combine method), could be changed to throwing exceptions upon incorrect colours.size